src/init.cpp

```diff
@@ -214,8 +216,9 @@ void Shutdown(NodeContext& node)
     StopTorControl();
 
     // After everything has been shut down, but before things get flushed, stop the
-    // CScheduler/checkqueue threadGroup
+    // CScheduler/checkqueue, threadGroup and load block thread.
     if (node.scheduler) node.scheduler->stop();
+    g_load_block.join();
```

Comment:
I think it might be safer to check if the thread is joinable before joining in case this path is reached after some initialization failure when the thread may not have started.

Reply:
Thanks, I've added that.